docs/source/usage.rst

```diff
@@ -305,6 +305,21 @@ Input yaml file:
     >>> print(conf.user.home)
     /home/omry
 
+It is possible to set a default value if the environment variable is missing. 
+In the following example, we set '1234' as the default value if the environment
+variable 'DB_PASSWORD' is not defined.
+
+.. doctest::
+
+    >>> cfg = OmegaConf.create({
+    ...       'database': {'password': '${env:DB_PASSWORD,12345}'}
+    ... })
+    >>> print(cfg.database.password)
+    12345
+    >>> cfg = cfg.copy()  # clear resolver catch
```

Comment:
That's a hack and the fact it works is a bug, use set_cache:

```python
OmegaConf.set_cache(cfg, {}) # clear resolver cache
```

Reply:
Thanks, was looking after a method like that. But get an error, with the following code: 

```
from omegaconf import OmegaConf
import os

cfg = OmegaConf.create({
      'database': {'password': '${env:DB_PASSWORD,12345}'}
})
print(cfg.database.password)

OmegaConf.set_cache(cfg, {}) # clear resolver cache
os.environ["DB_PASSWORD"] = 'secret'
print(cfg.database.password)
```

```
Traceback (most recent call last):
  File "test.py", line 11, in <module>
    print(cfg.database.password)
  File "C:\Users\kf9762\Desktop\tmp\omegaconf\omegaconf\dictconfig.py", line 182, in __getattr__
    return self.get(key=key, default_value=None)
  File "C:\Users\kf9762\Desktop\tmp\omegaconf\omegaconf\dictconfig.py", line 197, in get
    return self._resolve_with_default(
  File "C:\Users\kf9762\Desktop\tmp\omegaconf\omegaconf\basecontainer.py", line 65, in _resolve_with_default
    value = self._resolve_single(value) if isinstance(value, str) else value
  File "C:\Users\kf9762\Desktop\tmp\omegaconf\omegaconf\basecontainer.py", line 412, in _resolve_single
    return BaseContainer._resolve_value(root, match.group(1), match.group(2))
  File "C:\Users\kf9762\Desktop\tmp\omegaconf\omegaconf\basecontainer.py", line 394, in _resolve_value
    value = resolver(root_node, inter_key)
  File "C:\Users\kf9762\Desktop\tmp\omegaconf\omegaconf\omegaconf.py", line 278, in caching
    cache = OmegaConf.get_cache(config)[name]
KeyError: 'env'
```

Reply:
Found solution. Have to use `OmegaConf.set_cache(cfg, defaultdict(dict, {}))`

Reply:
Propose to make an own clear catch method #175

Reply:
Done